spinalcordtoolbox/scripts/sct_label_utils.py

```diff
@@ -374,8 +375,19 @@ def launch_sagittal_viewer(img: Image, labels: Sequence[int], msg: str, previous
 
     params.subtitle = msg
 
+    # NB: `num_points` is used as the limit for how many points *should* be in the image.
+    # If this number is exceeded, `SagittalController` will throw a `TooManyPoints` warning.
+    # To determine what `num_points should be, we consider two cases:
+    # - Case 1: `previous_points` is set: This means we're adding OR correcting existing labels.
+    #            Presumably, we're dealing with all unique labels. So, we just take the length
```

Comment:
My thought process was:

- Trying to update duplicate labels is _hard_ given our current codebase. It was making my head hurt. (Exactly as you've identified.)
- Adding duplicate labels is a use-case that only came up in the context of the (internal to NeuroPoly) compression project. It isn't something users have asked for. We only really need this feature to provide an SCT-only tutorial, rather than asking users to download FSLeyes. So, I made the (hopeful, optimistic, less-work) presumption that users aren't going to try to correct duplicate labels, and only follow what the tutorial shows. (This could very well be a faulty assumption!)

So, I very much took the easy way out here. I'm very thankful you've pointed this out, though.

Reply:
That's ok! Thanks for sharing your thought process. And like I said, this seems to already be working for the internal use-case, so it's not urgent to get a solid UI.